openjdk/barriers/mmtkObjectBarrier.hpp

```diff
@@ -11,6 +11,13 @@
 #include "../mmtkBarrierSetC1.hpp"
 #include "../mmtkBarrierSetC2.hpp"
 
+#define MMTK_ENABLE_OBJECT_BARRIER_FASTPATH true
+
+#define SIDE_METADATA_BASE_ADDRESS 0x0000060000000000L
```

Comment:
This is hardcoded based on https://github.com/mmtk/mmtk-core/blob/a1cda07a572385d25084ad595ae302da888d8bf3/src/util/side_metadata/constants.rs#L7. Any chance we can use the constant from mmtk-core?

@javadamiri Does this need to be changed with your new side metadata implementation?

Reply:
> We use `slowdebug` in our CI. Is the fastpath tested in our CI?

Yes fast-path is always included. But I suggest to use fastdebug in our CI to speed up the test time. fastdebug builds enabled all assertions as well.

Reply:
Actually all fast-path are hard-coded right now. The fast-path IR only performs correctly under the current MetadataSpec settings. I'm planning to pass the `MetadataSpec` instance to vm-bindings in the future, so VM bindings can generate correct fast-path (as well as these constants) without hardcoding anything.

Reply:
> > We use `slowdebug` in our CI. Is the fastpath tested in our CI?
> 
> Yes fast-path is always included. But I suggest to use fastdebug in our CI to speed up the test time. fastdebug builds enable all assertions as well.

I changed the build to fastdebug in https://github.com/mmtk/mmtk-openjdk/pull/52, as we will start testing 3 plans since that PR - it is worth spending a bit more time in building and saving time for running tests.

Reply:
> Actually all fast-path are hard-coded right now. The fast-path IR only performs correctly under the current MetadataSpec settings. I'm planning to pass the `MetadataSpec` instance to vm-bindings in the future, so VM bindings can generate correct fast-path (as well as these constants) without hardcoding anything.

OK. I think we should have @javadamiri to look at this PR as well, as he will change  the side metadata implementation in mmtk-core. It would be better if he knows what kind of assumptions/hardcode we have for the side metadata implementation in this binding PR.

Reply:
The base metadata address changes to `HEAP_END` in the new implementation, we will need to change this constant.
It also doesn't use chunks (e.g. `CHUNK_MASK`).
Overall, I don't think much change is required on the binding side, except for the fastpath IR which will become more similar to the Wenyu's own implementation.

Reply:
Sounds good. I will start merging the PR in mmtk-core and this PR.